### PR TITLE
Hook up feedback api

### PR DESCRIPTION
**Pull request by jstefaniuk-d2l**
Todo:

- [x] Fix demo page

- [x] Add Docs

- [x] Get SauceLabs creds

### CODE REVIEW

**Comment by jstefaniuk-d2l** [comment]
@dbatiste thanks! I'll take a look into those things

File: user-feedback-text-input.js [hunk 1]
```diff
@@ -54,9 +55,14 @@ class UserFeedbackTextInput extends LitElement {
 	}
 
 	serialize() {
+		const prepend = (text) => {
```
Comment (jstefaniuk-d2l) [inline user-feedback-text-input.js]: yes this is duplicated, no I don't feel like making a "prepend" mixin for such a small bit of code

**Comment by jstefaniuk-d2l** [comment]
@dbatiste I put the core backdrop in and all is good 👍 

The dialog-mixin seems to have a lot of functionality that the flyout doesn't necessarily need, so I may leave that out for now and re-evaluate at a later time since things are currently working pretty well

File: user-feedback-hm-interface.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+	// TODO: when there is an easier way for scripts to hook into the entity store, use that instead of this
+	async makeCall(href, { method = 'GET', body } = {}) {
+		if (!href) {
+			throw new Error('no href provided');
+		}
+		if (typeof body === 'object') {
+			body = JSON.stringify(body);
+		}
+		const response = await window.fetch(href, {
```
Comment (ctwomey1) [inline user-feedback-hm-interface.js]: wouldn't you want to be using d2lfetch here?

File: user-feedback-hm-interface.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+			method,
+			body: body,
+			headers: {
+				Authorization: this.token
```
Comment (ctwomey1) [inline user-feedback-hm-interface.js]: for most of our components, this.token could be a token getter function or a token. 

If this component is meant for wider consumption you need to make sure that you take that into consideration.  Even portfolio i believe works this way

Comment (ctwomey1) [reply user-feedback-hm-interface.js]: derp right

**Pull request merged**